test/stdtypes/concurrency_test.go

```diff
@@ -0,0 +1,51 @@
+package stdtypes
+
+import (
+	"io/ioutil"
+	"sync"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+func TestConcurrentTextMarshal(t *testing.T) {
```

Comment:
Ok Excellent can we make sure that this one test is run with the race flag when we run the travis job.